### PR TITLE
[#1336] Extract span context from incoming MQTT message

### LINKED ISSUE #1336: Support providing a trace context along with MQTT messages
In order to have a complete overview of a telemetry/event message lifecycle in a tracing solution, it would be good to have the message related trace spans created in a protocol adapter be combined with trace spans created in the originating device/gateway. For that, the message reaching the protocol adapter must have trace context information and the protocol adapter must use these.

The HTTP adapter already supports having a OpenTracing span context parameter provided via the HTTP headers of an incoming message.

A corresponding feature should be added to the MQTT adapter, providing support for extracting trace context information from a "property bag" at the end of the MQTT topic name.

(This relates to #267 in that there we also need support for "property bags".)

### PR DESCRIPTION
**Pull request by calohmn**
This fixes #1336.

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/tracing/TracingHelper.java [hunk 1]
```diff
@@ -308,6 +312,71 @@ public static SpanContext extractSpanContext(final Tracer tracer, final MultiMap
...
+     * This provides a generic way to serialize a span context in any kind of textual data.
+     * See {@link #extractSpanContext(Tracer, Supplier)} for the corresponding method to deserialize the
+     * context from that data.
+     *
+     * @param tracer The Tracer to use for injecting the context.
+     * @param spanContext The context to inject.
+     * @param keyValueConsumer The operation that will receive the key-value pairs representing the context.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public static void injectSpanContext(final Tracer tracer, final SpanContext spanContext,
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/tracing/TracingHelper.java]: where is this used?

Comment (calohmn) [reply core/src/main/java/org/eclipse/hono/tracing/TracingHelper.java]: Currently nowhere. But that method makes sense as the counterpart to the `extractSpanContext(tracer, supplier)` method and may be useful in external code using `TracingHelper`.

Comment (sophokles73) [reply core/src/main/java/org/eclipse/hono/tracing/TracingHelper.java]: ok, just wanted to make sure I'm not missing something ;-)

**Comment by sophokles73** [comment]
@calohmn do you want to document this new feature in the release notes?

**Comment by calohmn** [comment]
@sophokles73 As noted in #1336, the use cases for this feature are somewhat special. Therefore I don't think we need to mention it in the release notes.

**Pull request merged**